src/compass/s1_geocode_stack.py

```diff
@@ -135,6 +139,30 @@ def generate_burst_map(zip_files, orbit_dir, output_epsg=None, bbox=None,
     burst_map = pd.DataFrame(data=burst_map)
     return burst_map
 
+def _get_pol_str(zip_file, pol_type):
+    """Get the polarization string from the zip file name and the type of polarization."""
+    # S1A_IW_SLC__1SDV_20150224T114043_
+    # File name format:
+    # MMM_BB_TTTR_LFPP_YYYYMMDDTHHMMSS_YYYYMMDDTHHMMSS_OOOOOO_DDDDDD_CCCC.EEEE
+    # PP: Polarization: SH (single HH), SV (single VV), DH (dual HH+HV), DV (dual VV+VH)
+    from pathlib import Path
+    pp = Path(zip_file).stem[14:16]
+    count = pp[0]
+    h_or_v = pp[1]
+    if count == "S" and pol_type in ["dual-pol", "cross-pol"]:
+        raise ValueError(f"{zip_file} is single-pol. Unable to process {pol_type}")
+    if pol_type == "dual-pol":
+        raise NotImplementedError("TODO")
+        if h_or_v == "H":
+            return ["hh", "hv"]
+        else:
+            return ["vv", "vh"]
+    elif pol_type == "cross-pol":
+        return 'hv' if h_or_v == "H" else "vh"
+    elif pol_type == "co-pol":
+        return 'hh' if h_or_v == "H" else "vv"
+    else:
+        raise ValueError(f"Invalid {pol_type = }")
```

Comment:
```suggestion
    if pol_type not in ["dual-pol", "cross-pol", "co-pol"]:
        raise ValueError(f"Invalid {pol_type = }")
    if count == "S" and pol_type in ["dual-pol", "cross-pol"]:
        raise ValueError(f"{zip_file} is single-pol. Unable to process {pol_type}")
    if pol_type == "dual-pol":
        raise NotImplementedError("TODO")
        if h_or_v == "H":
            return ["hh", "hv"]
        else:
            return ["vv", "vh"]
    if pol_type == "cross-pol":
        return "hv" if h_or_v == "H" else "vh"
    if pol_type == "co-pol":
        return "hh" if h_or_v == "H" else "vv"
```

Early returns for readability